### PR TITLE
Semana 12

### PR DESCRIPTION
**Pull request by edirlucasi7**

### CODE REVIEW

**Review by thaisandre — APPROVED** [review]

**Comment by thaisandre** [comment]
muito bom, Edir! bem vindo ao time :)

File: src/main/webapp/WEB-INF/views/category/formUpdateCategory.jsp [hunk 1]
```diff
@@ -1,15 +1,8 @@
...
-    <head>
-        <meta charset="utf-8">
-        <link rel="stylesheet" type="text/css" href="/resources/css/style.css">
-        <link rel='stylesheet' href='/webjars/bootstrap/3.3.7/css/bootstrap.min.css'>
-        <title>Atualiza Categoria</title>
-    </head>
-    <body>
+<%@ taglib prefix="templates" tagdir="/WEB-INF/tags/templates" %>
 
+<templates:admin-template-header title="Atualiza Categoriria" />
```
Comment (mariasantosdev) [inline src/main/webapp/WEB-INF/views/category/formUpdateCategory.jsp]: Atualiza Categoria**

File: src/test/java/br/com/levelup/aluratech/validator/category/CheckCategoryUniqueCodeForEditionFormValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                .code("programacao")
+                .build();
+
+        validator.validate(updateCategoryRequest, errors);
+
+        verify(errors).rejectValue("code", "category.code.exists");
+    }
+
+    @Test
+    public void when_code_exists_and_id_is_equal_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/category/CheckCategoryUniqueCodeForEditionFormValidatorTest.java:35]: when_code_exists_and_id_is_equal_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/category/CheckCategoryUniqueCodeForEditionFormValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                .name("programacao")
+                .build();
+
+        validator.validate(updateCategoryRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_code_not_exists_for_id_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/category/CheckCategoryUniqueCodeForEditionFormValidatorTest.java]: when_code_not_exists_for_id_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/category/CheckCategoryUniqueCodeForEditionFormValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                .code("business")
+                .build();
+
+        validator.validate(updateCategoryRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_code_not_exists_and_id_is_different_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/category/CheckCategoryUniqueCodeForEditionFormValidatorTest.java:63]: when_code_not_exists_and_id_is_different_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueCodeForEditionFormValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                .code("programacao")
+                .build();
+
+        validator.validate(updateCourseRequest, errors);
+
+        verify(errors).rejectValue("code", "course.code.exists");
+    }
+
+    @Test
+    public void when_code_exists_and_id_is_equal_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueCodeForEditionFormValidatorTest.java:36]: when_code_exists_and_id_is_equal_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueCodeForEditionFormValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                .code("programacao")
+                .build();
+
+        validator.validate(updateCourseRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_code_not_exists_for_id_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueCodeForEditionFormValidatorTest.java]: when_code_not_exists_for_id_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueCodeForEditionFormValidatorTest.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                .code("business")
+                .build();
+
+        validator.validate(updateCourseRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_code_not_exists_and_id_is_different_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueCodeForEditionFormValidatorTest.java:64]: when_code_not_exists_and_id_is_different_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueNameForEditionFormValidatorTest.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+                .name("Programação")
+                .build();
+
+        validator.validate(updateCourseRequest, errors);
+
+        verify(errors).rejectValue("name", "course.name.exists");
+    }
+
+    @Test
+    public void when_name_exists_and_id_is_equal_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueNameForEditionFormValidatorTest.java:36]: when_name_exists_and_id_is_equal_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueNameForEditionFormValidatorTest.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+                .name("Programação")
+                .build();
+
+        validator.validate(updateCourseRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_name_not_exists_for_id_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueNameForEditionFormValidatorTest.java:51]: when_name_not_exists_for_id_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueNameForEditionFormValidatorTest.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                .name("Business")
+                .build();
+
+        validator.validate(updateCourseRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_name_not_exists_and_id_is_different_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/course/CheckCourseUniqueNameForEditionFormValidatorTest.java:64]: when_name_not_exists_and_id_is_different_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueCodeForEditionFormValidatorTest.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                .code("programacao")
+                .build();
+
+        validator.validate(updateSubCategoryRequest, errors);
+
+        verify(errors).rejectValue("code", "subcategory.code.exists");
+    }
+
+    @Test
+    public void when_code_exists_and_id_is_equal_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueCodeForEditionFormValidatorTest.java]: when_code_exists_and_id_is_equal_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueCodeForEditionFormValidatorTest.java [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+                .code("business")
+                .build();
+
+        validator.validate(updateSubCategoryRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_code_not_exists_and_id_is_different_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueCodeForEditionFormValidatorTest.java:64]: public void when_code_not_exists_and_id_is_different_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueNameForEditionFormValidatorTest.java [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+                .name("Programação")
+                .build();
+
+        validator.validate(updateSubCategoryRequest, errors);
+
+        verify(errors).rejectValue("name", "subcategory.name.exists");
+    }
+
+    @Test
+    public void when_name_exists_and_id_is_equal_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueNameForEditionFormValidatorTest.java]: when_name_exists_and_id_is_equal_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueNameForEditionFormValidatorTest.java [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+                .name("Programação")
+                .build();
+
+        validator.validate(updateSubCategoryRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_name_not_exists_for_id_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueNameForEditionFormValidatorTest.java:51]: when_name_not_exists_for_id_not_should_not_return_error

File: src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueNameForEditionFormValidatorTest.java [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+                .name("Business")
+                .build();
+
+        validator.validate(updateSubCategoryRequest, errors);
+
+        verify(errors, never()).rejectValue(anyString(), anyString());
+    }
+
+    @Test
+    public void when_name_not_exists_and_id_is_different_not_should_return_error() {
```
Comment (mariasantosdev) [inline src/test/java/br/com/levelup/aluratech/validator/subcategory/CheckSubCategoryUniqueNameForEditionFormValidatorTest.java:64]: when_name_not_exists_and_id_is_different_should_not_return_error

**Review by mariasantosdev — APPROVED** [review]

**Comment by mariasantosdev** [comment]
Muito bom Edir!! Bem vindoooo :rocket:

**Pull request merged**